From 1eb8bcb1fa6a0e278b9f8d5d83110e324beabae9 Mon Sep 17 00:00:00 2001
From: Eric Lundby <elundby@anaconda.com>
Date: Tue, 15 Sep 2026 20:10:00 +0000
Subject: [PATCH 25/25] ssl module: use -arm64 suffix for OpenSSL DLLs on win-arm64

Upstream hardcodes the -x64 suffix for OpenSSL DLL names
(libcrypto-3-x64.dll / libssl-3-x64.dll), but conda-forge
OpenSSL on win-arm64 ships libcrypto-3-arm64.dll /
libssl-3-arm64.dll, so MYSQL_CHECK_SSL_DLLS fails with
"Cannot find SSL dynamic libraries".
---
 cmake/ssl.cmake | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/cmake/ssl.cmake b/cmake/ssl.cmake
index 8559a02..adcbf1f 100644
--- a/cmake/ssl.cmake
+++ b/cmake/ssl.cmake
@@ -803,15 +803,18 @@ MACRO(MYSQL_CHECK_SSL_DLLS)
       # OpenSSL 1.1 Look for libcrypto-1_1-x64.dll or libcrypto-1_1.dll
       # OpenSSL 1.0 Look for libeay32.dll
       SET(SSL_MSVC_VERSION_SUFFIX)
-      SET(SSL_MSVC_ARCH_SUFFIX)
+      IF(CMAKE_SYSTEM_PROCESSOR MATCHES "ARM64" OR
+         MSVC_CXX_ARCHITECTURE_ID MATCHES "ARM64")
+        SET(SSL_MSVC_ARCH_SUFFIX "-arm64")
+      ELSE()
+        SET(SSL_MSVC_ARCH_SUFFIX "-x64")
+      ENDIF()
       IF(OPENSSL_VERSION_MAJOR VERSION_EQUAL 1 AND
          OPENSSL_VERSION_MINOR VERSION_EQUAL 1)
         SET(SSL_MSVC_VERSION_SUFFIX "-1_1")
-        SET(SSL_MSVC_ARCH_SUFFIX "-x64")
       ENDIF()
       IF(OPENSSL_VERSION_MAJOR VERSION_EQUAL 3)
         SET(SSL_MSVC_VERSION_SUFFIX "-3")
-        SET(SSL_MSVC_ARCH_SUFFIX "-x64")
       ENDIF()
 
       FIND_FILE(HAVE_CRYPTO_DLL
-- 
2.39.5

